File: hackerRank/algorithms/MarsExploration/MarsExploration.java
package algorithms.MarsExploration;

import java.util.stream.IntStream;

// https://www.hackerrank.com/challenges/mars-exploration/problem
public class MarsExploration {
    static int marsExploration(String s) {
        String msg = "SOS";
        return (int)IntStream.range(0, s.length())
                .filter(i -> s.charAt(i) != msg.charAt(i % 3))
                .count();
    }

    public static void main(String[] args) {
        System.out.println(marsExploration("SOSSPSSQSSOR"));    //3
        System.out.println(marsExploration("SOSSOT"));          //1
        System.out.println(marsExploration("SOSSOSSOS"));       //0
    }
}

File: hackerRank/algorithms/MarsExploration/MarsExploration.js
// https://www.hackerrank.com/challenges/mars-exploration/problem
function marsExploration(s) {
    let pattern = "SOS";
    return [...Array(s.length).keys()]
        .filter((i) => s[i] != pattern[i % 3])
        .length;
}

console.log(marsExploration("SOSSPSSQSSOR"));    //3
console.log(marsExploration("SOSSOT"));          //1
console.log(marsExploration("SOSSOSSOS"));